### PR TITLE
docs: update contributing.md

### PR DESCRIPTION
**Pull request by zmsn-2077**
## Description

1. Referring to PyTorch's [CONTRIBUTING.md](https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md), I have reworked the CONTRIBUTING.md OmniSafe.
2. `make format` meet the issue:github.com/google/addlicense@latest: cannot use path@version syntax in GOPATH mode, solve it in #229.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/OmniSafeAI/omnisafe/issues) for new features and bug fixes)
#226 

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -41,16 +41,57 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
...
 **Please run `make format`** to reformat your code. You can check the codestyle using `make check-codestyle` and `make lint`.
 
-Please document each function/method and [type](https://google.github.io/pytype/user_guide.html) them using the following template:
+Please document each function/method and [type](https://google.github.io/pytype/user_guide.html) them using the following template, which is similar to the [PyTorch docs style]:
+
+Similar to the standard PyTorch Style docstring formatting rules, the following guidelines should be followed for docstring types (docstring types are the type information contained in the round brackets after the variable name):
+
+- The `Callable`, `Any`, `Iterable`, `Iterator`, `Generator` types should have their first letter capitalized.
+
+- The `list` and `tuple` types should be completely lowercase.
```
Comment (Gaiejj) [inline CONTRIBUTING.md:50]: Extra
```python
from __future__ import annotations
```
is needed for Python 3.8 to use ``tuple``, ``list``.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,16 +41,57 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
...
+    Args:
+        my_var (int, optional): Variable description. Default: 1
+        ...
 
-    :param arg1: describe what is arg1
-    :param arg2: describe what is arg2
-    :return: describe what is returned
+    Returns:
+        return_var (int, optional): Variable description.
+        ...
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:94]:
```suggestion
    Returns:
        Variable description.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -41,16 +41,57 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
...
+
+```python
+x (type1 or type2): Variable description.
+y (type1, type2, or type3): Variable description.
+```
+
+Please document each function/method and [type](https://google.github.io/pytype/user_guide.html) them using the following template, which is similar to the [PyTorch Docstring]():
 
 ```python
 def my_function(arg1: type1, arg2: type2) -> returntype:
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:83]:
```suggestion
def my_function(arg1: type1, arg2: type2, my_var: int = 1) -> returntype:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -75,23 +116,24 @@ To run tests with `pytest`:
...
+make pylint
+make mypy
 ```
 
 Codestyle check with `black`, `isort` and `flake8`:
 
 ```bash
-make check-codestyle
-make lint
+make py-format
```
[the remark this answers is not in the document]
Comment (zmsn-2077) [reply CONTRIBUTING.md:129]: solve the issue in #229

**Review by Gaiejj — APPROVED** [review]
LGTM.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -41,16 +41,58 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
...
+Please document each function/method and [type](https://google.github.io/pytype/user_guide.html) them using the following template, which is similar to the [PyTorch Docstring]():
+
+```python
+def my_function(arg1: type1, arg2: type2, my_var: int = 1) -> returntype:
+    """Short description of the function.
+
+    (Optional) Long description of the function or example usage.
+
+    Args:
+        my_var (int, optional): Variable description. Default: 1
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:91]:
```suggestion
        arg1 (type1): Variable description.
        arg2 (type2): Variable description.
        my_var (int, optional): Variable description. Default: 1
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -108,8 +150,7 @@ make spelling
 
 ## Changelog and Documentation
 
-Please do not forget to update the changelog (`docs/misc/changelog.rst`) and add documentation if needed.
+Please do not forget to update the [CHANGELOG.md](https://github.com/OmniSafeAI/omnisafe/blob/main/CHANGELOG.md) and add documentation if needed.
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:153]:
```suggestion
Please do not forget to update the [CHANGELOG.md](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CHANGELOG.md) and add documentation if needed.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -41,16 +41,58 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
 
 **Please run `make format`** to reformat your code. You can check the codestyle using `make check-codestyle` and `make lint`.
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:42]:
```suggestion
**Please run `make format`** to reformat your code. You can check the codestyle using `make lint`.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -41,16 +41,58 @@ We are using [black codestyle](https://github.com/psf/black) (max line length of
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:41]: max line length of 100

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -29,28 +27,71 @@ git clone https://github.com/OmniSafeAI/omnisafe
 cd omnisafe/
 ```
 
-2. Install OmniSafe in develop mode, with support for building the docs and running tests:
+1. Install OmniSafe in develop mode, with support for building the docs and running tests:
```
Comment (XuehaiPan) [inline CONTRIBUTING.md:30]:
```suggestion
2. Install OmniSafe in develop mode, with support for building the docs and running tests:
```

**Review by XuehaiPan — APPROVED** [review]

**Pull request merged**